feat(codecov): Add AI page and improve coverage UI

Adal3n3 opened the pull request:
## Summary

This PR adds a new Prevent AI page to the Codecov (Prevent) section and improves the coverage UI with enhanced functionality.
<img width="1533" height="572" alt="Screenshot 2025-07-28 at 11 58 30 AM" src="https://github.com/user-attachments/assets/db10fe61-ab91-481b-8d4d-1969ff6d5a79" />

<img width="1533" height="773" alt="Screenshot 2025-07-28 at 11 58 52 AM" src="https://github.com/user-attachments/assets/ae839264-8f86-4dfc-8014-b1920ade93a9" />


## Changes
- 🤖 Add new AI page with aiWrapper component and tests
- 🛣️ Update routes to include AI functionality
- 📊 Enhance coverage page UI and add commits view
- 🧭 Update codecov secondary navigation
- 🎨 Add prevent-ai illustration asset
- ⚙️ Improve settings page integration

## Files Added/Modified
- `static/app/views/codecov/ai/ai.tsx` - Main AI page component
- `static/app/views/codecov/ai/aiWrapper.tsx` - AI wrapper component
- `static/app/views/codecov/ai/ai.spec.tsx` - AI page tests
- `static/app/views/codecov/ai/aiWrapper.spec.tsx` - AI wrapper tests
- `static/app/views/codecov/coverage/coverage.tsx` - Improved coverage page
- `static/app/views/codecov/coverage/commits.tsx` - Added commits functionality
- `static/app/routes.tsx` - Added AI route configuration
- `static/images/prevent-ai-illustration.svg` - New illustration asset

## Testing
- ✅ All new components include comprehensive test coverage
- ✅ ESLint and Stylelint checks passed
- ✅ TypeScript compilation successful
- ✅ Prettier formatting applied

Ready for review! 🚀

Adal3n3 committed:
feat(codecov): Add AI page and improve coverage UI

- Add new AI page with aiWrapper component and tests
- Update routes to include AI functionality
- Enhance coverage page UI and add commits view
- Update codecov secondary navigation
- Add prevent-ai illustration asset
- Improve settings page integration

Adal3n3 committed:
feat(codecov): Add AI page and improve coverage UI

- Add new AI page with aiWrapper component and tests
- Update routes to include AI functionality
- Enhance coverage page UI and add commits view
- Update codecov secondary navigation
- Add prevent-ai illustration asset
- Improve settings page integration

evanpurkhiser commented on static/app/routes.tsx:
```diff
@@ -2000,6 +2000,12 @@ function buildRoutes() {
           />
         </Route>
       </Route>
+      <Route path="ai/">
+        {/* Render AI page with layout wrapper */}
+        <Route component={make(() => import('sentry/views/codecov/ai/aiWrapper'))}>
+          <IndexRoute component={make(() => import('sentry/views/codecov/ai/ai'))} />
+        </Route>
+      </Route>
```
you'll need to rebase, I just converted these all over to object style routes

evanpurkhiser commented on static/app/views/codecov/ai/ai.tsx:
```diff
@@ -0,0 +1,156 @@
+              {t('• It generates unit tests for untested code in your PR.')}
+            </BulletItem>
+          </BulletList>
+          <LearnMoreLink
```
Any reason this can't use ExternalLink?

Adal3n3 replied:
Updated.

evanpurkhiser commented on static/app/views/codecov/ai/ai.tsx:
```diff
@@ -0,0 +1,156 @@
+          <Description>
+            {t('Prevent AI is a generative AI agent that automates tasks in your PR:')}
+          </Description>
+          <BulletList>
```
I think we should have a reusable component for lists you can use

Adal3n3 replied:
Agree, happy to use it when we have it.

evanpurkhiser replied:
Sorry, I meant we do have it: https://github.com/getsentry/sentry/blob/31baa2ff4dd30f2f92844017433f4daee918b545/static/app/components/list/index.tsx

evanpurkhiser commented on static/app/views/codecov/ai/ai.tsx:
```diff
@@ -0,0 +1,156 @@
+          <IllustrationImage src={preventAiIllustration} alt="PreventAI illustration" />
+        </ImageColumn>
+        <TextColumn>
+          <Title>{t('Ship Code That Breaks Less With Code Reviews And Tests')}</Title>
+          <Description>
+            {t('Prevent AI is a generative AI agent that automates tasks in your PR:')}
+          </Description>
```
I think @JonasBa just added some new text primitives that maybe can be used here

Adal3n3 replied:
Can you add more colors here on what to use?

evanpurkhiser commented on static/app/views/codecov/coverage/commits.tsx:
```diff
@@ -14,3 +708,340 @@ const LayoutGap = styled('div')`
+`;
+
+const TooltipDescription = styled('div')`
+  font-weight: 400;
+  font-size: 12px;
```
There's theme properties for font sizes and boldness

evanpurkhiser commented on static/app/views/codecov/coverage/coverage.tsx:
```diff
@@ -91,7 +91,7 @@ const SAMPLE_DATA = {
   return (
     <LayoutGap>
-      <p>Coverage Analytics</p>
+      <p>Code Coverage</p>
```
Looks like we're missing translations here in general `t('Code Coverage')`

evanpurkhiser commented:
I don't think the sentry Frontend is in a place where we can support PRs generated by AI to this degree. There's a lot of inconsistency with how we write frontend views in this PR that will need to be addressed before we merge something like this

Adal3n3 commented:
@evanpurkhiser I will work on the changes and update this pr.

Adal3n3 committed:
Address reviewer feedback: Fix translations, font styles, and null safety

- Add missing translation for 'Code Coverage' text
- Replace hardcoded font-weight and font-size with theme properties
- Add null safety checks for uploadCount to prevent TypeError
- Make route structure consistent with existing codecov patterns

Adal3n3 committed:
Address reviewer feedback: Fix translations, font styles, and null safety

- Replace hardcoded font-weight and font-size with theme properties in AI page
- Add translation wrappers for upload status text (Processed, Pending, Failed)
- Add null safety checks for uploadCount to prevent TypeError
- Ensure consistent styling using Sentry's design system

Adal3n3 commented on static/app/views/codecov/ai/ai.tsx:
```diff
@@ -0,0 +1,156 @@
+          <BulletList>
+            <BulletItem>
+              {t(
+                '• It reviews your pull request, predicting errors and suggesting code fixes.'
```
Updated

Adal3n3 committed:
fix(codecov): Complete reviewer feedback implementation

- Add comprehensive translations for all UI text elements
- Replace hardcoded font styles with theme properties
- Implement robust null safety for uploadCount properties
- Ensure consistent styling with Sentry design system

Zylphrex committed:
style(explore): Simplify chart loading states (#96433)

This simplifies the chart loading states by using a transparent loading
mask when there's already data in chart and only showing the spinner on
first load.

cathteng committed:
fix(aci): skip funneling events through workflow engine for ignored issues (#96594)

scttcper committed:
fix(onboarding): Remove trailing slash after hash (#96595)

No slashes at the end of the url hash

taylor-osler-sentry committed:
Merge branch 'master' into at/add_prevent_ai_page_and_coverage_UI

aliu39 committed:
ref(feedback): pass project obj to create_feedback_issue (#96562)

Shouldn't cause changes in function

In all `create_feedback_issue` callsites (except `save_event_feedback`),
we have the project object, but pass in its ID only to query for project
again at the top of the fx. This changes the fx to take in the object so
we can reuse it. Saves a query and reuses cached relations like
`project.organization`.

`save_event_feedback` must take an ID since it's a task, so all params
have to be serializable.

Also moves metric/logs out of `should_filter_feedback` and adds project
platform to some metrics (in DD we'd need to turn on aggregations
though)

taylor-osler-sentry committed:
fix merged route

michellewzhang committed:
ref(replay): default tab is breadcrumbs again (#96601)

lobsterkatie committed:
feat(grouping): Add `get` method to `GroupingContext` class (#96598)

The `GroupingContext` class we use as a datastore during event grouping has a dictionary-like interface, but doesn't support `.get()` the way actual dictionaries do. This adds the method, which will allow us to stop setting default values at the top level of our grouping configs when they're only needed by certain strategies. (For example, see `variant`, `is_recursion`, and `exception_data` in the base config[1].)

[1] (https://github.com/getsentry/sentry/blob/510c723483d6eaed11fb307fc7cd1c17ec451a26/src/sentry/grouping/strategies/configurations.py#L30-L47

malwilley committed:
fix(aci): Allow partial updates in detector PUT endpoint (#96596)

We often want to update just a portion of the detector, like when
enabling/disabling it. This updates the PUT endpoint to use
`partial=True` and updates the validator `update()` function so that
it's compatible with that.

iamrajjoshi committed:
:wrench: chore: update logs to debug activity task (#96614)

NicoHinderling committed:
chore(launchpad): Ensure sentry secret is picked up from the environment (#96616)

<!-- Describe your PR here. -->

<!--

  Sentry employees and contractors can delete or ignore the following.

-->

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated
in the State of Delaware in 2015 as Functional Software, Inc. and is
gonna need some rights from me in order to utilize my contributions in
this here PR. So here's the deal: I retain all rights, title and
interest in and to my contributions, and by keeping this boilerplate
intact I confirm that Sentry can use, modify, copy, and redistribute my
contributions, under Sentry's choice of terms.

priscilawebdev committed:
ref: Update tempest logic to look for playstation in enabled console platforms (#96619)

closes
https://linear.app/getsentry/issue/TET-958/update-tempest-logic-to-check-for-playstation-in

obostjancic committed:
fix(agents-insights): models table pagination (#96531)

nsdeschenes committed:
chore(search-bar): Add key suggestion feature flag (#96545)

Add in feature flag for some upcoming work around suggestion `key:value`
pairs based off of the raw input value.

Ticket: EXP-376

priscilawebdev committed:
feat: Add private gaming sdk access modal to getting started docs (#96622)

Add Private Gaming SDK Access Modal to the playstation,xbox and
nintendo-switch Getting Started Docs

priscilawebdev committed:
feat: Add Nintendo Switch 2 (#96620)

closes
https://linear.app/getsentry/issue/TET-963/add-nintendo-switch-2-to-frontend

Nintendo Switch 2 reuses everything from Nintendo Switch. The only
differences are the name and logo.

priscilawebdev committed:
feat: Add analytics to gaming modals (#96625)

closes https://linear.app/getsentry/issue/TET-914/add-analytics

priscilawebdev committed:
ref: Replace PlaystationSdkAccessModal with PrivateGamingSdkAccessModal (#96623)

shellmayr committed:
ref(analytics): move CheckinProcessingErrorStored to typed event (#96512)

JonasBa committed:
text: remove color html attrib (#96608)

Color is a native html attribute typed as just a string.

armenzg committed:
fix(deletions): If there are any more groups keep chunking (#96526)

The assumption is that we schedule 100 groups for deletion and there
should be none left, however, this is not the case.

Instead of raising an error, let's keep deleting and keep logging it.

Fixes [SENTRY-48W4](https://sentry.sentry.io/issues/6770253794/)

---------

Co-authored-by: Mark Story <mark@mark-story.com>

priscilawebdev committed:
fix: Error type in RequestSdkAccessButton (#96638)

constantinius committed:
ref(analytics): Transform analytics events for issue workflows (#95213)

constantinius committed:
ref(analytics): Transform analytics events for SnubaQuery (#95205)

adrianviquez committed:
Add testsuites hook (#96494)

This PR adds the test suite FE hook to fetch test suite related
information.

Notes
-
- Adds an API call to get test suites in the `TestSuiteDropdown`
component
- Adds the `useTestSuites` hook
- Adjusts `useGetTestResults` hook to further filter by `testSuites` if
selected
- Lift `useInfinitedTestResults` hook to it's parent component to
leverage defaultBranch field - hook is only enabled when all orgId,
repo, branch and date are selected, so this avoids unnecessary,
preemptive calls
- Conditionally renders summary + test suites components 
- Adjusts context value to reset URL params for `testSuites` when repo +
org are changed

lzhao-sentry committed:
feat(dashboards): add cell actions to `TableWidgetVisualization` (#96524)

### Changes

Adds props to `TableWidgetVisualization` to support custom cell actions
in the table. Also updates the story and unit tests. No UI change, as it
is currently not being used by tables yet

Zylphrex committed:
ref(explore): Use shared visualize and group by components in explore… (#96572)

… spans

This updates explore spans to use the shared components with the span
specific customizations on top. Once the page params are merged, most of
the customizations should go away as well.

enochtangg committed:
fix(taskworker): Add unavailable host tag for taskworker metric (#96646)

iamrajjoshi committed:
:mag: nit: swallow timeout errors for issue summary (#96649)

vgrozdanic committed:
ref(hybridcloud): Refactor queries on cacheversion tables (#96627)

Lately `hybridcloud_regioncacheversion` has been quite hot, and one of
the reasons for it could be inefficient queries performed.

This PR optimizes query executed on the table, from (worst case
scenario) total of 4 to 2 queries. Before we were using our custom
`create_or_update` method, instead of Django's `update_or_create` which
is slightly better and more optimized, and before we used to perform
extra SELECT after we have updated the value, which caused additional
load on the database without the need, since we already had the object
in memory.


Since this is a critical path, the change will be rolled out gradually
in 10-50-100 percentages.

AbhiPrasad committed:
fix(onboarding): Show logs in gatsby onboarding docs (#96651)

Missed this during https://github.com/getsentry/sentry/pull/96425

markstory committed:
fix(hybridcloud) Continue to track down write volume from sentryapps (#96587)

Continue the work started in #96440 to track down why we're getting
higher than expected write volumes to regioncache.

Based on the writes coming from the sentryapp + product workers and not
RPC web pods, my next theory is that we're getting invalid/incomplete
data back from cache and hitting this delete.

iamrajjoshi committed:
:wrench: chore: fix some typing for gitlab issues spec (#96645)

asottile-sentry committed:
ref: fix types for sentry.integrations.utils.sync (#96420)

<!-- Describe your PR here. -->

k-fish committed:
ref(ourlogs): Refactor the code file path renderer to avoid n+1 (#96575)

### Summary
Adding this as a column would cause a lot of headaches for our
integration that fetches code links. This PR also cleans up and adds
more tests to the log table row spec.

asottile-sentry committed:
ref: fix types for project_rules_configuration endpoint (#96422)

<!-- Describe your PR here. -->

cmanallen committed:
fix(replays): Use aggregate search config in bulk delete job (#96589)

Before aggregate filters were causing the query to fail.

rbro112 committed:
chore(emerge): Leverage new Flex, Text & Heading components for Emerge build details page (#96574)

Leverages new Flex, Text & Heading components on the Emerge build
details page:

Before:
<img width="1492" height="666" alt="Screenshot 2025-07-28 at 1 03 56 PM"
src="https://github.com/user-attachments/assets/7c76e854-71ce-43d5-8be1-3e8501f7335a"
/>

After (no change as expected):
<img width="1490" height="643" alt="Screenshot 2025-07-28 at 1 01 50 PM"
src="https://github.com/user-attachments/assets/63c00429-99dc-4a90-a21e-1aaf57477f99"
/>

asottile-sentry committed:
ref: strong typing for sentry.utils.query (#95982)

<!-- Describe your PR here. -->

adrianviquez committed:
feat(Codecov): Get rid of main branch default value (#96652)

This PR gets rid of the main default value for branch query parameter on
the test_results endpoint as a) this endpoint does and can support
`None` as a value and b) that was a placeholder value during testing

getsentry-bot committed:
Revert "fix(hybridcloud) Continue to track down write volume from sentryapps (#96587)"

This reverts commit 27785a282a9c2c233024103ff9f71f0561a5471a.

Co-authored-by: asottile-sentry <103459774+asottile-sentry@users.noreply.github.com>

AbhiPrasad committed:
fix(js-sdk): Use correct `enableLogs` flag (#96654)

We are dogfooding the v10 alpha:


https://github.com/getsentry/sentry/blob/82f640be370f447b29539928c904cafadba981a9/package.json#L71

but forgot to change the `enableLogs` flag.

See
https://github.com/getsentry/sentry-javascript/blob/develop/MIGRATION.md#2-removed-apis

phacops committed:
chore(eap): Remove option to switch entities for EAP subscriptions (#96607)

We do not need this option anymore.

Zylphrex committed:
feat(logs): Return hasLogs in projects response (#96591)

calvin-codecov committed:
feat(codecov): Add TA onboarding steps for steps 4,5,6 (#96599)

Closes https://linear.app/getsentry/issue/CCMRG-1437/onboarding-step-4
Closes https://linear.app/getsentry/issue/CCMRG-1438/onboarding-step-5
Closes https://linear.app/getsentry/issue/CCMRG-1439/onboarding-step-6

Figma file:
https://www.figma.com/design/mSyCX0wCk7e2yItQfiMhaa/GH-1247_MergePlan?node-id=1-2&t=pUq7qhgXR70GkbJr-1

- Added new steps that are used for steps 4, 5, and 6. None currently
have the step 6 label but depending on how many steps came before, the
ViewResultsInsightsStep may be used for 6.
- Added "Step" suffix to all of the step component names
- Changed a couple instances of `space` to the new schema with theme

Commit:
chore(trace-view-vitals): refactor pill styling (#96477)

- Refactor code for pills in trace view

**Before:**
<img width="942" height="182" alt="Screenshot 2025-07-25 at 1 46 53 PM"
src="https://github.com/user-attachments/assets/4cc4c556-3c1f-4c87-b4f4-cc37844b06db"
/>
<img width="508" height="143" alt="Screenshot 2025-07-25 at 1 49 43 PM"
src="https://github.com/user-attachments/assets/1e59bc89-87a4-449a-a24e-e7180b31b480"
/>

**After:**
<img width="946" height="175" alt="Screenshot 2025-07-25 at 1 46 39 PM"
src="https://github.com/user-attachments/assets/a0a23946-e392-4b62-bddc-d98272811bd2"
/>
<img width="507" height="142" alt="Screenshot 2025-07-25 at 1 49 22 PM"
src="https://github.com/user-attachments/assets/108c13b7-650f-42a1-9280-ad5701b84748"
/>

shayna-ch committed:
feat(issues): add subtitles for common search filters (#96382)

kcons committed:
fix(aci): More manageable delayed_workflow logs (#96611)

We can't log the raw redis data; it's too big, too noisy, and it yields
broken JSON we can't query.
This replaces that with two slightly more targeted data log messages to
let us definitively identify involved workflows, and their associated
groups/events.
We prefer sorted lists over sets because sets log unordered and are hard
to search through.

Also, this moves from logging DataConditionGroups (which have a large
and not particularly useful string repr) to just the sorted IDs.

evanpurkhiser committed:
ref(rr6): Remove RouteWithName export (#96563)

iamrajjoshi committed:
:wrench: chore(aci): logs to check if we are correctly invoking the activity notification handler (#95735)

vishnupsatish committed:
feat(feedback): label generation at ingest, stored as tags (#96390)

scttcper committed:
feat(rr6): Add deprecatedRouterProps to route objects (#96588)

This uses typescript to enforce that only some routes are injected with
the legacy router props. This way we can slowly reduce the components
that require this react router 3 style prop.

Removes the "NoOp" route component and renders undefined instead which
react router 6 should just render the child?

---------

Co-authored-by: Evan Purkhiser <evanpurkhiser@gmail.com>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

malwilley committed:
fix(aci): Reject condition_group for error detectors (#96602)

cvxluo committed:
ref(nav-v2): removes nav v2 flag  (#95955)

Nav V2 is GAed and available for everyone (see
[automator](https://github.com/getsentry/sentry-options-automator/blob/507cffc82679fc5aee370808a4b084dc6a416d27/options/default/flagpole.yml#L3538)),
so we don't need the flag. Follow up to
https://github.com/getsentry/sentry/pull/95951.

AbhiPrasad committed:
feat(onboarding): Add logs to ruby onboarding (#96653)

resolves
https://linear.app/getsentry/issue/LOGS-259/add-logs-to-ruby-onboarding

<img width="1514" height="759" alt="image"
src="https://github.com/user-attachments/assets/8cd30708-3715-43dc-9576-372720450416"
/>

AbhiPrasad committed:
feat(onboarding): Add logs to ruby/rails onboarding (#96671)

resolves
https://linear.app/getsentry/issue/LOGS-260/add-logs-to-ruby-rails-onboarding

roggenkemper committed:
chore(detectors): Update span for performance detection span sorting  (#96661)

we only care about the actual sorting here - the flagpole call was
adding some noise. also adds a better description for the span

evanpurkhiser committed:
ref(uptime): Remove unused uptimeSubscriptionId (#96670)

michellewzhang committed:
fix(replay): fix navigation timestamp in log (#96669)

<img width="640" height="20" alt="SCR-20250729-jqrk"
src="https://github.com/user-attachments/assets/660428c3-22bb-4dec-912c-4fd14f07baee"
/>

gggritso committed:
fix(discover): Top N queries do not support grouping by array fields (#96032)

- Fix a bug where top events did not correctly work with array fields, and instead just ignored their existence entirely

AbhiPrasad committed:
feat(onboarding): Add logs to ruby-rack onboarding (#96673)

resolves
https://linear.app/getsentry/issue/LOGS-261/add-logs-to-ruby-rack-onboarding

scttcper committed:
fix(issues): Prevent "Set up code mapping" from closing frame (#96675)

kcons committed:
perf(aci): Make get_latest_release_for_env cache not-found releases (#96674)

We frequently see slow tasks from repeated latest release condition
evaluations.
This may be due to the cache not being used when releases aren't found.
Adds caching and a metric to observe how much value it is adding.

Also adds a typed cache access helper to clarify None vs False, which is
a bit overkill for a single use but helpful nonetheless.

cathteng committed:
chore(auth-v2): record analytics for rotating csrf token (#96677)

lzhao-sentry committed:
fix(dashboards): fix tooltip in certain fields not aligned to cell text (#96656)

### Changes

[Merged a PR](https://github.com/getsentry/sentry/pull/96323) a few days
ago that fixed the vertical alignment the cells, but it messed up the
alignment of the tooltip. This is to fix that while still persisting the
fix for the vertical alignment

### Before
<img width="324" height="145" alt="Screenshot 2025-07-29 at 11 48 08 AM"
src="https://github.com/user-attachments/assets/b6e1e080-90b1-4244-a4e3-9f8bfaf91e16"
/>


### After
<img width="356" height="137" alt="Screenshot 2025-07-29 at 11 48 19 AM"
src="https://github.com/user-attachments/assets/ee2c5cc3-2b24-4090-a918-112f9e9bd322"
/>

cvxluo committed:
chore(issues): remove option registration from inferring project platform (#96678)

Follow up from
https://github.com/getsentry/sentry-options-automator/pull/4702,
cleaning up leftover option from rolling out inferring project platforms
from https://github.com/getsentry/sentry/pull/95402

asottile-sentry committed:
ref: fix typing for fixtures.schema_validation (#96667)

this decorator caused anything decorated to become Any

<!-- Describe your PR here. -->

JonasBa committed:
ref(storybook): add primitives section  (#96515)

Adds a new 'primitives' category to the storybook navigation tree to
group
fundamental layout and text components separately from other core
components.

- Add 'primitives' to StoryCategory type
- Update category ordering to place primitives after foundations
- Create isPrimitivesFile() to identify core/layout and core/text
components
- Modify inferFileCategory() to categorize primitives components
correctly

This improves component organization by creating a dedicated section for
the most basic UI building blocks.

wmak committed:
test(explore): Add tests for the explore spans page (#96579)

- Making sure that we have at least a base level of testing of the
explore page
- depends on https://github.com/getsentry/sentry/pull/96578

asottile-sentry committed:
ref: convert ProjectOption.value to django JSONField (#96682)

<!-- Describe your PR here. -->

JonasBa committed:
grid: add grid implementation (#96597)

Add Grid layout primitive that follows the Flex conventions

jarrettscott committed:
fix(billing): Change display name to Logs (#96660)

Closes:
https://linear.app/getsentry/issue/BIL-1178/logs-product-should-display-in-the-ui-as-logs-not-log-bytes

Changes the display name for the Logs product. Affects the subscription
card (shown below) as well as other labels in the application.

<img width="1178" height="343" alt="Screenshot 2025-07-29 at 9 53 02 AM"
src="https://github.com/user-attachments/assets/557821cd-998a-48f1-be60-c3ca6c13e102"
/>

ameliahsu committed:
feat(aci): set up detector Ongoing Issues (#96604)

<img width="856" height="694" alt="Screenshot 2025-07-29 at 9 47 43 AM"
src="https://github.com/user-attachments/assets/606673d9-f62a-4c71-8a35-bfdef1ee1bc3"
/>

extracted time selector for metric detector chart to the details
component so that it could be shared between the chart and table

Zylphrex committed:
ref(multi-query): Use shared chart visualization in multi query mode (#96666)

This refactors the multi query mode to use the same shared chart
visualization as other parts of explore.

lobsterkatie committed:
ref(grouping): Always use default config when loading unknown config (#96610)

Currently, we have inconsistent behavior when loading a grouping config with an id we don't recognize. If the config id comes from the project, and we've therefore gotten it by using `ProjectGroupingConfigLoader._get_config_id`, it's guaranteed to be one we've heard of, because if it's not, `_get_config_id` will instead return the id of the default config[1]. If instead the config id comes from an event's `grouping_config` entry (which we grab with no validation[2]), trying to load the grouping config will raise a `GroupingConfigNotFound` error[3].

To make the behavior consistent, this switches from raising the error in the latter case to just returning the default config. Since this was the only place we were raising `GroupingConfigNotFound`, this also lets us remove all of our handling of it, as well as the error class itself.


[1] https://github.com/getsentry/sentry/blob/65fa0a9af7d186bc5142cf85e6fdb9a59e08a406/src/sentry/grouping/api.py#L147-L148
[2] https://github.com/getsentry/sentry/blob/65fa0a9af7d186bc5142cf85e6fdb9a59e08a406/src/sentry/grouping/api.py#L190
[3] https://github.com/getsentry/sentry/blob/65fa0a9af7d186bc5142cf85e6fdb9a59e08a406/src/sentry/grouping/api.py#L229-L230

getsentry-bot committed:
ci(jest): regenerate jest-balance.json (#95512)

This PR was auto-generated - it updates the \`jest-balance.json\` file
with new test run data from CI.

---------

Co-authored-by: JonasBa <JonasBa@users.noreply.github.com>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

getsentry-bot committed:
Revert "ref: convert ProjectOption.value to django JSONField (#96682)"

This reverts commit 055056db270e20f28cba50de631a66ebbb91fe80.

Co-authored-by: asottile-sentry <103459774+asottile-sentry@users.noreply.github.com>

markstory committed:
Reapply "fix(hybridcloud) Continue to track down write volume from sentryapps v2 (#96657)

This reverts commit
https://github.com/getsentry/sentry/commit/51eed24e9ae938ed8387f1e2c281f564848ebeaa.

Redo of https://github.com/getsentry/sentry/pull/96587 which was
reverted because of semantic conflicts.

scttcper committed:
fix(discover): Refresh discover saved queries on delete/duplicate (#96685)

just refreshes the current query

this acceptance test below is testing this case, however it currently
only passes because it attempts to navigate to the saved query directly
after deleting (event propagation) therefore the card disappears and it
passes.


https://github.com/getsentry/sentry/blob/195d6990b8a694ee733a1483e0b3f7486b53a5f4/tests/acceptance/test_organization_events_v2.py#L447-L472

roggenkemper committed:
chore(rate-limit): Remove legacy rate limit pages  (#96354)

these pages are only accessible by orgs who had the legacy feature but
we don't want any orgs to be able to use this anymore. After we remove
the pages from the front end, we will clean up the backend.

fixes
https://linear.app/getsentry/issue/RTC-1091/remove-rate-limit-pages-from-the-front-end

scttcper committed:
fix(insights): Add legacy router props to database span summary (#96697)

ryan953 committed:
ref(settings): Refactor settings/$project/toolbar/ to not rely on deprecatedRouteProps (#96695)

scttcper committed:
ref(performance): Remove trace wrapper props (#96703)

both `ExploreContent` and `Content` do not take props. Remove "any"

scttcper committed:
ref(u2f): Add missing prop types to u2fEnrolledDetails (#96700)

scttcper committed:
ref(ui): Add missing div props, remove any (#96702)

ryan953 committed:
ref(settings): Refactor settings/$project/replay to not rely on deprecatedRouteProps (#96701)

scttcper committed:
ref(alerts): Add prop types to alert ErrorChart (#96705)

remove any prop type

kddubey committed:
fix(severity): Add option to route severity requests to GPU (#95992)

as part of [migration
plan](https://www.notion.so/sentry/Moving-severity-and-fixability-services-to-GPU-2348b10e4b5d80688a64e3235a653744?source=copy_link#2348b10e4b5d800aa55dccb0695ed2ba)
to serve severity on `group-ingest` pods

need to wait for https://github.com/getsentry/ops/pull/16321 before
increasing option value from 0

scttcper committed:
feat(issues): Remove legacy props from issue details (#96706)

Finishes the migration away from legacy router props for all issue
details pages

scttcper committed:
ref(insights): Add prop types to database alert banner (#96704)

Remove any

malwilley committed:
feat(aci): Disable monitor creating/editing when user does not have alerts:write (#96687)

Adal3n3 commented:
I'm going to keep this pr open for prototyping purposes so I don't need to work on production codes. I created a new branch called branch: adalene/main, moving forward I will merge my prototyping prs into my own branch so I can use the Vercel Preview link for others to interact with the prototype. This branch: at/add_prevent_ai_page_and_coverage_UI has been rebased onto a child of adalene/main.

JoshFerge reviewed (changes requested):
- Please split up frontend and backend changes
- This PR is far far too large to be able to be reviewed. 

See https://develop.sentry.dev/engineering-practices/code-review/#guidelines-forsubmitters for more information. Thank you!

JoshFerge commented:
ah i see that this is likely due to a bad rebase -- would be best to close this PR and re-open a new one.

Adal3n3 commented:
@JoshFerge I'm so sorry... you are right this bad rebase is a midnight for me. I will take your advise close this pr, and create a new one.
